preprolanguage/language/src/main/java/com/oracle/truffle/prepro/parser/PreProNodeFactory.java

```diff
@@ -466,7 +466,18 @@ public PreProExpressionNode createNumericLiteral(Token literalToken) {
     }
 
     public PreProExpressionNode createExistsExpression(Token nameToken) {
-        final PreProExistsNode preProExistsNode = new PreProExistsNode(nameToken.getText());
+        final String text = nameToken.getText();
```

Comment:
@orat:
Wir waren uns beim Exists builtin nicht mehr ganz sicher, was genau wir mit dem Exists builtin überprüfen sollten?
Die Anwesenheit einer Variable des Host systems (also, ob ein `import("name")` funktioniert), oder ob eine Variable im Code definiert wurde?

Reply:
Grundsätzlich ist es so dass wir typischerweise "extern" Sätze mit viele Zeitreihen haben, von denen nicht jeder Satz alle Zeitreihen enthält. Alle diese Sätze sollen mit dem gleiche propro-code prozessiert werden. Der Grund dafür ist, dass es zu viele Varianten geben kann. Also ganz konkret es fehlt aus messtechnischen Gründen immer mal wieder eine Zeitreihe aber immer mal wieder eine andere. Daher muss man im prepro-Programm überprüfen können ob eine Zeitreihe vorhanden ist und je nachdem ob verschiedene Code-Fragmente aufrufen. Also z.B. wenn die Zeitreihe eines bestimmten Markers nicht vorhanden ist, dann wird statt dessen der Mittelwert von zwei anderen Zeitreihen verwendet. Bei der ersten Studienarbeit gab es ja noch einen "globalen" import, in der graal-vm-Version muss man ja nun jede zeitreihe einzeln importieren, was auch nicht gerade optimal ist. Ich weiss jetzt nicht mehr was passiert wenn der Import fehlschlägt. Eigentlich sollte es dann nur eine Warnung geben. Dann macht es nämlich Sinn mit exist(Zeitreihe) Überprüfen zu können ob eine Zeitreihe überhaupt vorhanden ist. Ein exist() das das Vorhandensein einer in prepro-definierten Variablen überprüft ist meiner Meinung nach sinnlos, denn das ist ja dann ein Programmierfehler, oder? In diesem Fall sollte also sowieso das Abarbeiten des Programms abgebrochen werden.